helpers/Physics.h

```diff
@@ -0,0 +1,321 @@
+} PhysicsConstraint;
+
+typedef struct {
+	s32   num;  // How many limbs will be smoothed with push
+	Vec3f push; // direction Z, pushes based on rot[0]
+	f32   mult; // How much the pushing fill affect
+	Vec3f rot;  // DEG, rigids towards, relative rot
+} PhysicsRigidity;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32 limbsLength[];
+} PhysicsStrandInit;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32* limbsLength;
+} PhysicsStrand;
+
+_Z64HDR_HELPER_PREFIX_
+void Physics_GetHeadProperties(PhysicsStrand* strand, Vec3f* mult, s32 flag) {
+	static MtxF mtxF;
+	Vec3f zero = { 0 };
+	
+	Matrix_Get(&mtxF);
+	strand->head.mtxF = &mtxF;
+	func_800D20CC(&mtxF, &strand->head.rot, flag);
+	if (mult == NULL) {
+		Matrix_MultVec3f(&zero, &strand->head.pos);
+		
+		return;
+	}
+	Matrix_MultVec3f(mult, &strand->head.pos);
+}
+
+_Z64HDR_HELPER_PREFIX_
+void Physics_SetPhysicsStrand(PhysicsStrandInit* init, PhysicsStrand* dest, f32* lengthDest, Vec3f* sphereCenters) {
+	dest->info = init->info;
+	dest->head = init->head;
+	dest->gfx = init->gfx;
+	dest->spheres = init->spheres;
+	dest->constraint = init->constraint;
+	dest->rigidity = init->rigidity;
+	
+	for (s32 i = 0; i < init->info.numLimbs + 1; i++) {
+		lengthDest[i] = init->limbsLength[i];
+	}
+	dest->limbsLength = lengthDest;
+	dest->spheres.centers = sphereCenters;
+}
+
+_Z64HDR_HELPER_PREFIX_
+void Physics_DrawDynamicStrand(GraphicsContext* gfxCtx, TwoHeadGfxArena* disp, PhysicsJoint* jointTable, PhysicsStrand* strand, void* callback, void* callbackArg1, void* callbackArg2) {
+	s32 i;
+	f32 tempY;
+	f32 angX;
+	f32 angY;
+	Vec3f workVec;
+	Vec3f posAdd;
+	Vec3f rigidity, smoothedRigid;
+	Vec3f velAdj = { 0 };
+	s16 headRotY = strand->head.rot.y;
+	Vec3f* pPos;
+	Vec3f* pRot;
+	Vec3f* pVel;
+	Vec3f* pPrevPos;
+	Vec3f* pPrevRot;
+	
+	Matrix_Push();
+	
+	jointTable[0].pos.x = strand->head.pos.x;
+	jointTable[0].pos.y = strand->head.pos.y;
+	jointTable[0].pos.z = strand->head.pos.z;
+	
+	for (i = 1; i < strand->info.numLimbs + 1; i++) {
+		Math_ApproachF(&jointTable[i].vel.x, 0.0f, 1.0f, strand->info.velStep);
+		Math_ApproachF(&jointTable[i].vel.y, 0.0f, 1.0f, strand->info.velStep);
+		Math_ApproachF(&jointTable[i].vel.z, 0.0f, 1.0f, strand->info.velStep);
+	}
+	
+	Matrix_RotateX_s(strand->head.rot.x, MTXMODE_NEW);
+	Matrix_RotateY_s(strand->head.rot.y, MTXMODE_APPLY);
+	Matrix_RotateZ_s(strand->head.rot.z, MTXMODE_APPLY);
+	if (strand->rigidity.rot.x) Matrix_RotateX_f(strand->rigidity.rot.x, MTXMODE_APPLY);
+	if (strand->rigidity.rot.y) Matrix_RotateY_f(strand->rigidity.rot.y, MTXMODE_APPLY);
+	if (strand->rigidity.rot.z) Matrix_RotateZ_f(strand->rigidity.rot.z, MTXMODE_APPLY);
```

Comment:
```suggestion
	if (strand->rigidity.rot.x != 0) Matrix_RotateX_f(strand->rigidity.rot.x, MTXMODE_APPLY);
	if (strand->rigidity.rot.y != 0) Matrix_RotateY_f(strand->rigidity.rot.y, MTXMODE_APPLY);
	if (strand->rigidity.rot.z != 0) Matrix_RotateZ_f(strand->rigidity.rot.z, MTXMODE_APPLY);
```
Imo only not check against a value for something that's meant as a boolean